consensus/taiko/consensus.go

```diff
@@ -229,12 +229,14 @@ func (t *Taiko) FinalizeAndAssemble(chain consensus.ChainHeaderReader, header *t
 
 	// Verify anchor transaction
 	if len(body.Transactions) != 0 { // Transactions list might be empty when building empty payload.
-		isAnchor, err := t.ValidateAnchorTx(body.Transactions[0], header)
```

Comment:
@mask-pp Doesn't this already check if the first transaction is an anchor transaction?

Reply:
If the calling link contains `miner/miner.go:Pending()` function and `txpool.Pending` is not empty, then return error(`ErrAnchorTxNotFound`).
We know it's not not normal.